### PR TITLE
NUP-2342: Update examples to use capnp serialization

### PR DESCRIPTION
**Pull request by lscheinkman**
@scottpurdy Please review, it depends on https://github.com/numenta/nupic.core/pull/1355

### CODE REVIEW

File: examples/bindings/sparse_matrix_how_to.py [hunk 1]
```diff
@@ -96,10 +95,22 @@
...
-s2 = cPickle.load(open('sm.txt', 'rb'))
-print '\nPickling\n', s2
+# A sparse matrix can be serialized:
+schema = SM32.getSchema()
+
+# Save
+with open("sm.bin", "wb") as f:
+  proto = schema.new_message()
+  s.write(proto)
+  proto.write(f)
```
Comment (scottpurdy) [inline examples/bindings/sparse_matrix_how_to.py]: If you open the file in"w+b" mode then you can do "f.seek(0)" after write and then read from the same file. Doesn't really matter though

File: examples/bindings/svm_how_to.py [hunk 2]
```diff
@@ -167,9 +166,19 @@ def persistence():
...
 
-    print "Pickling dense classifier"
-    cPickle.dump(classifier, open('test', 'wb'))
-    classifier = cPickle.load(open('test', 'rb'))
+    print "Serializing dense classifier"
+
+    # Save
+    schema = classifier.getSchema()
+    with open("test", "wb") as f:
+        proto = schema.new_message()
```
Comment (scottpurdy) [inline examples/bindings/svm_how_to.py]: Two space indent?

Comment (lscheinkman) [inline examples/bindings/svm_how_to.py]: I used 4 spaces to be consistent with the rest of the file

Comment (scottpurdy) [inline examples/bindings/svm_how_to.py]: Kk makes sense

**Comment by lscheinkman** [comment]
@scottpurdy please merge if you approve

**Pull request merged**